build.sh

```diff
@@ -13,22 +14,25 @@ usage()
   echo "  providing no target is the same as 'core plugins gradle maven'"
   echo "  -q  quick mode: skip tests, formatting"
   echo "  -e  show error information (mvn: -e, gradle: --stacktrace --info)"
+  echo "  -n  dry run: show what will be performed"
   exit 1
 }
 
 # doBuild: Run a command in a directory
 # $1 = directory
 # $2... = build command
 doBuild() {
-  (directory="$1"; shift; echo ">>> (cd $directory; $*)"; cd "$directory" && eval '"$@"')
+  (directory="$1"; shift; echo ">>> (cd $directory; $*)"; \
+    if [ -z "$dryRun" ]; then cd "$directory" && eval '"$@"'; fi)
```

Comment:
I don't think everything should fit in a single line in parentheses.

```sh
directory="$1"; shift; echo ">>> (cd $directory; $*)
if ... ; then (cd "$directory" && eval '"$@"') fi
```

Reply:
Why?  Without the parens means we expose `directory`.

I could simplify this to `[ -z "$dryRun" ] && cd "$directory" && eval '"$@"'`

Reply:
I mean, not removing the parentheses, but they don't have to wrap the echo part, so it becomes like `then (cd "$directory" && eval '"$@"') fi`.

or
```sh
[ -z "$dryRun" ] && (cd "$directory" && eval '"$@"')
```

Reply:
But outer parens are required for `directory` [to be assigned as a local](https://stackoverflow.com/questions/18597697/posix-compliant-way-to-scope-variables-to-a-function-in-a-shell-script).

Reply:
Okay, either is fine. It's just that the previous code didn't show great readability to me. FYI, if your intention is to have the local scope of `directory`, our Google style guide is to use the keyword `local`, which the Stack Overflow also says is a usual choice. Anyways, if we used the `if` clause, I would have it like below for better readability.

```sh
  (
    directory="$1"; shift; echo ">>> (cd $directory; $*)
    if ... then; (cd "$directory" && eval '"$@"') fi
  )
```